internal/core/ngt/ngt.go

```diff
@@ -277,18 +281,27 @@ func (n *ngt) Search(vec []float32, size int, epsilon, radius float32) ([]Search
 	if radius == 0 {
 		radius = n.radius
 	}
+	cvec := (*C.float)(&vec[0])
+	defer C.free(unsafe.Pointer(cvec))
+	csize := *(*C.size_t)(unsafe.Pointer(&size))
+	defer C.free(unsafe.Pointer(&csize))
+	cepsilon := *(*C.float)(unsafe.Pointer(&epsilon))
+	defer C.free(unsafe.Pointer(&cepsilon))
+	cradius := *(*C.float)(unsafe.Pointer(&radius))
+	defer C.free(unsafe.Pointer(&cradius))
 
 	n.mu.RLock()
 	ret := C.ngt_search_index_as_float(
 		n.index,
-		(*C.float)(&vec[0]),
+		// *(*C.float)(unsafe.Pointer(&vec[0])),
```

Comment:
?

Reply:
?

Reply:
i thought you've forgot to delete these comments